Assets/Scripts/UI/Indicator/TargetIndicator.cs

```diff
@@ -0,0 +1,177 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class TargetIndicator : MonoBehaviour
+{
+    public Sprite humanIndicator, ghostIndicator;
+    
+    public Image TargetIndicatorImage;
+    public Image OffScreenTargetIndicator;
+    public float OutOfSightOffset = 20f;
+    private float outOfSightOffest { get { return OutOfSightOffset /* canvasRect.localScale.x*/; } }
+
+    private GameObject target;
+    private Camera mainCamera;
+    private RectTransform canvasRect;
+
+    private RectTransform rectTransform;
+
+    private void Awake()
+    {
+        rectTransform = GetComponent<RectTransform>();
```

Comment:
I'm not sure if this is the same as typing "transform". Idk if you want to test to do transform instead of rectTransform if it works.

Reply:
Then we can avoid doing GetComponent<T>() calls